lib/danger_plugin.rb

```diff
@@ -86,6 +88,29 @@ def output_report(report)
       if !threshold.nil? && (report.coverage * 100) < threshold
         fail("Code coverage under minimum of #{threshold}%")
       end
+
+      # Notify failure if minimum coverage hasn't been reached for modified/added files
+      file_threshold = args.first[:minimum_coverage_percentage_for_changed_files].to_i
+      ignore_list = args.first[:filename_prefix_ignore_list]
+      if !file_threshold.nil?
+        report.targets.each do |target|
+          target.files.each do |file|
+            if ((file.coverage * 100) < file_threshold) #check if file coverage is less than that defined
+              if !ignore_list.nil? #check is ignore list is provided or not
+                ignore_list.each do |ignore|
+                  if (!file.name.include? ignore)
+                    fail("Class code coverage is below minimum. Improve #{file.name} to at least #{file_threshold}%")
+                  else
+                    puts("File #{file.name} is being skipped from minimum code coverage because it contains '#{ignore}' keyword.")
+                  end
+                end
+              else
+                fail("Class code coverage is below minimum, please improve #{file.name} to at least #{file_threshold}%")
+              end
+            end
+          end
+        end
+      end
```

Comment:
this deep nesting looks like code smell to me. you can achieve the same without the nesting:

```ruby
file_threshold = args.first[:minimum_coverage_percentage_for_changed_files].to_i || 0
ignore_list = args.first[:filename_prefix_ignore_list] || []

if file_threshold > 0
  report.targets.each do |target|
    target.files.each do |file|
      should_ignore = ignore_list.any? {|ignored| file.name.include? ignore }
      puts("Skipping minimum code coverage check for #{file.name} because it contains '#{ignore}' keyword.") if should_ignore
      if ((file.coverage * 100) < file_threshold && !should_ignore)
        fail("Class code coverage is below minimum, please improve #{file.name} to at least #{file_threshold}%")
      end
    end
  end
end
```

or with less logging:

```ruby
file_threshold = args.first[:minimum_coverage_percentage_for_changed_files].to_i || 0
ignore_list = args.first[:filename_prefix_ignore_list] || []

if file_threshold > 0
  report.targets.each do |target|
    target_files = target.files.select { |file| ignore_list.none? {|prefix| file.name.include? prefix } }
    violations = target_files.select { |file| file.coverage * 100) < file_threshold}
    fail("Class code coverage is below minimum, please improve #{violations.map {|f| f.name }} to at lea st #{file_threshold}%") if !violations.empty?
  end
end
```

Reply:
Thank you for this suggestion, I think less logging option makes sense to me and we can go with that.